### DISCUSSION TITLE
structured data for dataset's column headers

### DISCUSSION THREAD

**Discussion by bhuffaker** [schemaorg/schemaorg#3319] [Ideas]
I would like to add structured data for our dataset's column headers. What is the current state of https://lists.w3.org/Archives/Public/public-vocabs/2013Aug/att-0033/Lookinginsidetables.html? 

<img width="909" alt="image" src="https://user-images.githubusercontent.com/17103357/225161237-0ae72d73-8f30-4a46-97e8-47f5eaa6eb88.png">

**Comment by danbri** [comment]
You might consider W3C CSVW, if you only want column names / data types

**Comment by bhuffaker** [comment]
CSVW doesn't support annotating columns with class/type. I have a drafted an extension to CSVW here: https://catalog.caida.org/paper/2023_annotated_schema . Which supports the table diagramed above. 

I would be love feedback.

**Comment by danbri** [comment]
if I understand you correctly, CSVW's notion of virtual columns will let you add entity types.

For example, this [config](https://github.com/w3c/csvw/blob/gh-pages/experiments/browserdemo/eg11.json) maps columns into a few types, and adds virtual columns with the typing information.

You might also be interested in https://tarql.github.io/ and the tmcf format [as used](https://docs.datacommons.org/custom_dc/prepare_data.html) at datacommons.org

**Comment by bhuffaker** [comment]
Thanks.  I didn't understand that "valueURL" would hold the type/class annotation. I will have more time after the IMC '23 paper deadline this Friday, but right now I still don't know how to create the full set of annotations in the example above. 
- one of the main goals I would like to solve with these annotations is to help dataset uses find and understand object identifiers in the dataset.  From the example above, Social Security Numbers is a Person identifier. 
   - sometimes identifiers are created across multiple columns.  How can I annotate a multi-column property?  In the example below, the Person property name uses values from the first_name and given_name columns.
      <img width="571" alt="Screenshot 2023-05-25 at 9 25 22 AM" src="https://github.com/schemaorg/schemaorg/assets/17103357/5aec672e-baf0-4d90-b3a2-1b880d7222b4">
  - I would also like to annotate the identifier's namespace. So the region property would have identifiers using the ISO Alpha2 namespace. 
- There is also an alias problem when different individuals of the same class are in the same table. In the example above, the Employee and Manager individuals are both of type Person, but different individuals. The valueURL doesn't provide enough information to make that distinction.  Ideally there would be an annotation to provide this grouping.

**Comment by danbri** [comment]
It's a long time since I was immersed in the details, but one thing to say is that for CSVW (and many other) table-to-graph tools, there will usually be transformations that aren't possible (such as concat'ing strings, or conditionals). You might find SPARQL 1.1 adds enough here for you, eg. to apply SPARQL constructs downstream of the CSVW RDF transform, or by approaches like TARQL above.

**Comment by bhuffaker** [comment]
I looked into SPARQL, it adds to much complexity. We have a hard enough time getting grad & undergrads to annotated their datasets as it is.

**Comment by akuckartz** [comment]
Maybe RML can be used?  See https://rml.io/
